src/circular_buffer.jl

```diff
@@ -1,22 +1,31 @@
 """
-    CircularBuffer{T}(n)
+    CircularBuffer{T}(v,n)
 
 The CircularBuffer type implements a circular buffer of fixed capacity
 where new items are pushed to the back of the list, overwriting values
 in a circular fashion.
 
-Allocate a buffer of elements of type `T` with maximum capacity `n`.
+Allocate a buffer of elements of type `T`  containing items `v` with maximum capacity `n`.
 """
 mutable struct CircularBuffer{T} <: AbstractVector{T}
     capacity::Int
     first::Int
     length::Int
     buffer::Vector{T}
 
-    CircularBuffer{T}(capacity::Int) where {T} = new{T}(capacity, 1, 0, Vector{T}(undef, capacity))
+    function CircularBuffer{T}(vec,capacity::Int) where {T}
+        capacity >= length(vec) || throw(BoundsError()) # prevent overflow
+        new{T}(capacity,1,length(vec),copyto!(Vector{T}(undef,capacity),convert(Vector{T},vec)))
+    end
+
 end
 
-CircularBuffer(capacity) = CircularBuffer{Any}(capacity)
+CircularBuffer(capacity::Int) = CircularBuffer{Any}(capacity)
+
+CircularBuffer{T}(capacity::Int) where {T} = CircularBuffer{T}([],capacity)
```

Comment:
We can avoid a call to `convert` and an allocation here by intitalizing the buffer to empty of the right type
```suggestion
CircularBuffer{T}(capacity::Int) where {T} = CircularBuffer{T}(T[],capacity)
```